t/integration/test_batches.py

```diff
@@ -270,3 +270,30 @@ def acks(requests: List[SimpleRequest]) -> None:
 
     # After the tasks are done, both results are acked.
     assert acked == [result_1.id, result_2.id]
+
+
+def test_retry(celery_app: Celery, celery_worker: TestWorkController) -> None:
+    """The batch task fails on the first call, is retried and succeeds."""
+
+    def signal(sender: Union[Task, str], **kwargs: Any) -> None:
+        assert celery_app.current_task.name == "t.integration.tasks.retry_if_even"
+
+    # One success, one failure and successful retry
+    counter = SignalCounter(2, signal)
+    signals.task_success.connect(counter)
+
+    result_1 = retry_if_even.delay(1)
+    retry_if_even.delay(2)
+
+    # The flush interval is 1 second and the retry interval is 3 seconds,
+    # this is longer.
+    sleep(5)
```

Comment:
To match our other changes:

```suggestion
    # The flush interval is 0.1 seconds and the retry interval is 0.3 seconds,
    # this is longer.
    sleep(0.5)
```

Reply:
I had to mess with the timing a bit to get it to be consistently successful, but it's shorter than it was before.